volatility3/framework/plugins/windows/mftscan.py

```diff
@@ -0,0 +1,331 @@
+# This file is Copyright 2022 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+import logging
+
+from struct import unpack
+from typing import Dict
+
+from volatility3.framework import constants, renderers, interfaces
+from volatility3.framework.configuration import requirements
+from volatility3.framework.exceptions import PagedInvalidAddressException
+from volatility3.framework.objects import utility
+from volatility3.framework.renderers import conversion, format_hints
+from volatility3.plugins import yarascan
+
+vollog = logging.getLogger(__name__)
+
+try:
+    import yara
+except ImportError:
+    vollog.info("Python Yara module not found, plugin (and dependent plugins) not available")
+    raise
+
+signatures = {
+    'mft_objects': """rule mft_headers
+                                {
+                                strings:
+                                  $header1 = "FILE0"
+                                  $header2 = "FILE*"
+                                  $header3 = "BAAD"
+                                condition:
+                                  any of them
+                                }"""
+}
+
+# https://github.com/volatilityfoundation/volatility/blob/a438e768194a9e05eb4d9ee9338b881c0fa25937/volatility/plugins/mftparser.py#L60
+ATTRIBUTE_TYPE_ID = {
+    0x10:"STANDARD_INFORMATION",
+    0x20:"ATTRIBUTE_LIST",
+    0x30:"FILE_NAME",
+    0x40:"OBJECT_ID",
+    0x50:"SECURITY_DESCRIPTOR",
+    0x60:"VOLUME_NAME",
+    0x70:"VOLUME_INFORMATION",
+    0x80:"DATA",
+    0x90:"INDEX_ROOT",
+    0xa0:"INDEX_ALLOCATION",
+    0xb0:"BITMAP",
+    0xc0:"REPARSE_POINT",
+    0xd0:"EA_INFORMATION",  #Extended Attribute
+    0xe0:"EA",
+    0xf0:"PROPERTY_SET",
+    0x100:"LOGGED_UTILITY_STREAM",
+}
+
+VERBOSE_STANDARD_INFO_FLAGS = {
+    0x1:"Read Only",
+    0x2:"Hidden",
+    0x4:"System",
+    0x20:"Archive",
+    0x40:"Device",
+    0x80:"Normal",
+    0x100:"Temporary",
+    0x200:"Sparse File",
+    0x400:"Reparse Point",
+    0x800:"Compressed",
+    0x1000:"Offline",
+    0x2000:"Content not indexed",
+    0x4000:"Encrypted",
+    0x10000000:"Directory",
+    0x20000000:"Index view",
+}
+
+FILE_NAME_NAMESPACE = {
+    0x0:"POSIX", # Case sensitive, allows all Unicode chars except '/' and NULL
+    0x1:"Win32", # Case insensitive, allows most Unicide except specials ('/', '\', ';', '>', '<', '?')
+    0x2:"DOS",   # Case insensitive, upper case, no special chars, name is 8 or fewer chars in name and 3 or less extension
+    0x3:"Win32 & DOS", # Used when original name fits in DOS namespace and 2 names are not needed
+}
+
+MFT_FLAGS = {
+    0x0: "Removed",
+    0x1: "File", # "In Use",
+    0x2: "Directory", # if flag & 0x0002 == 0 this is a regular file
+    0x3: "Directory"
+}
+
+INDEX_ENTRY_FLAGS = {
```

Comment:
This one doesn't have a 0 defined, so again, there's inconsistency between how we deal with these two sets of flags.

Reply:
These were taken directly from the old MFT parser plugin in the same format 

https://github.com/volatilityfoundation/volatility/blob/a438e768194a9e05eb4d9ee9338b881c0fa25937/volatility/plugins/mftparser.py#L60


According to the spec on all the sources I can find there is no possible value for 0. We can add one but as its not a valid option it doesn't make sense?

Reply:
Yep, again, happy that other people do it a particular way, more just suggesting we don't need to do it that way if there's something else that makes more sense.

Specifically, if 0 isn't a valid option, what happens if we encounter it (because we were fed bad data).  How do we want to handle that situation?